akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala

```diff
@@ -36,32 +36,32 @@ private[akka] class DirectByteBufferPool(defaultBufferSize: Int, maxPoolEntries:
   private def allocate(size: Int): ByteBuffer =
     ByteBuffer.allocateDirect(size)
 
-  @tailrec
-  private final def takeBufferFromPool(): ByteBuffer =
-    if (locked.compareAndSet(false, true)) {
-      val buffer =
-        try if (buffersInPool > 0) {
+  private final def takeBufferFromPool(): ByteBuffer = {
+    var buffer: ByteBuffer = null
+
+    if (lock.tryLock(1, TimeUnit.MILLISECONDS))
```

Comment:
Why? As discussed extensively elsewhere on this PR we do _not_ want to just drop things to the floor if we cannot “immediately” get access to the pool, and a lock is the perfect and well-optimized way of expressing this concern. Removing locks is not self-justifying.